CMakeLists.txt

```diff
@@ -75,6 +81,142 @@ target_compile_definitions(notcurses
     FORTIFY_SOURCE=2 _GNU_SOURCE SOURCE_DEFAULT
 )
 
+# libnotcurses++
+set(NCPP_SOURCES
+  src/libcpp/NotCurses.cc
+  src/libcpp/PanelReel.cc
+  src/libcpp/Plane.cc
+  src/libcpp/Root.cc
+  src/libcpp/Tablet.cc
+  src/libcpp/Visual.cc
+  )
+
+add_library(
+  notcurses++
+  SHARED
+  ${NCPP_SOURCES}
+  )
+
+add_library(
+  notcurses++-static
+  STATIC
+  ${NCPP_SOURCES}
+  )
+set_target_properties(
+  notcurses++-static PROPERTIES
+  OUTPUT_NAME notcurses++
+  )
+
+set_target_properties(
+  notcurses++ PROPERTIES
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
+  OUTPUT_NAME "notcurses++")
+
+set(NCPP_INCLUDE_DIRS
+  include
+    "${PROJECT_BINARY_DIR}/include"
+    "${TERMINFO_INCLUDE_DIRS}"
+  )
+
+target_include_directories(notcurses++
+  PRIVATE ${NCPP_INCLUDE_DIRS}
+  )
+
+target_include_directories(notcurses++-static
+  PRIVATE ${NCPP_INCLUDE_DIRS}
+  )
+
+target_link_libraries(notcurses++
+  PRIVATE
+  notcurses)
+
+set(NCPP_COMPILE_OPTIONS
+  -Wall
+  -Wextra
+  -W
+  -Wshadow
+  -Wformat
+  -Werror=format-security
+  -Wnull-dereference
+  -Wmisleading-indentation
+  -Wunused
+  -Wpedantic
+  -Wsuggest-override
+  -Wno-c99-extensions
+  -fno-strict-aliasing
+  -ffunction-sections
+  -funswitch-loops
+  -finline-limit=300
+  -fstack-protector
+  -fno-rtti
+  -fno-exceptions
+  -mpopcnt
+  -fpic
+  -pthread
+  ${DEBUG_OPTIONS}
+  )
+
+set(NCPP_COMPILE_DEFINITIONS_PUBLIC
+  FORTIFY_SOURCE=2 _GNU_SOURCE SOURCE_DEFAULT
+  )
+set(NCPP_COMPILE_DEFINITIONS_PRIVATE
+  _XOPEN_SOURCE # wcwidth(3) requires _XOPEN_SOURCE, and is in our headers
+  )
+
+target_compile_options(notcurses++
+  PRIVATE
+  ${NCPP_COMPILLE_OPTIONS}
+  )
+
+target_compile_options(notcurses++-static
+  PRIVATE
+  ${NCPP_COMPILLE_OPTIONS}
+  )
+
+target_compile_definitions(notcurses++
+  PUBLIC
+    ${NCPP_COMPILE_DEFINITIONS_PUBLIC}
+  PRIVATE
+    ${NCPP_COMPILE_DEFINITIONS_PRIVATE}
+  )
+
+target_compile_definitions(notcurses++-static
+  PUBLIC
+    ${NCPP_COMPILE_DEFINITIONS_PUBLIC}
+  PRIVATE
+    ${NCPP_COMPILE_DEFINITIONS_PRIVATE}
+  )
+
+file(GLOB NCPP_HEADERS
+  CONFIGURE_DEPENDS
+  LIST_DIRECTORIES false
+  ${CMAKE_SOURCE_DIR}/include/ncpp/*.hh)
```

Comment:
.hh?!?! D: lol

Reply:
`.cc` and `.hh` :) But, to be honest, this is for `ccls` (it's a language server for `C/C++` projects) support in Emacs - it has an annoying issue of treating `.h` as C source which raises all kinds of red flags (even though I *do* specifically tell it that my `.h` is a C++ header, oh well)

Reply:
urk ok. i dread CMakeLists.txt expansion but i guess it's gonna happen.